frontend/simple/views/Invite.vue

```diff
@@ -98,22 +102,32 @@ export default {
       this.members.splice(index, 1)
     },
     submit: async function () {
-      for (let i = 0; i < this.members.length; i++) {
-        let member = this.members[i]
-        let events = await backend.eventsSince(member.contractId, member.contractId)
-        let [contract, ...actions] = events.map(e => {
-          return Events[e.entry.type].fromObject(e.entry, e.hash)
-        })
-        let state = contract.toVuexState()
-        actions.forEach(action => {
-          let type = action.constructor.name
-          contract.constructor.vuex.mutations[type](state, action.data)
-        })
-        let mailbox = await backend.latestHash(state.attributes.mailbox)
-        let invite = new Events.PostInvite({groupId: this.$store.state.currentGroupId}, mailbox)
-        await backend.publishLogEntry(state.attributes.mailbox, invite)
+      try {
+        this.errorMsg = null
+        for (let i = 0; i < this.members.length; i++) {
+          let member = this.members[i]
+          let events = await backend.eventsSince(member.contractId, member.contractId)
+          let [contract, ...actions] = events.map(e => {
+            return Events[e.entry.type].fromObject(e.entry, e.hash)
+          })
+          let state = contract.toVuexState()
+          actions.forEach(action => {
+            let type = action.constructor.name
+            contract.constructor.vuex.mutations[type](state, action.data)
+          })
+          let mailbox = await backend.latestHash(state.attributes.mailbox)
+          let date = new Date()
+          let invite = new Events.PostInvite({groupId: this.$store.state.currentGroupId, sentDate: date.toString()}, mailbox)
+          await backend.publishLogEntry(state.attributes.mailbox, invite)
+          let latest = await backend.latestHash(this.$store.state.currentGroupId)
+          let invited = new Events.RecordInvitation({ username: member.name, inviteHash: invite.toHash(), sentDate: date.toString() }, latest)
+          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
```

Comment:
This is something I thought about as well but I decided any solution to this problem at this point would be premature until we know all the business rules surrounding when we can and can't invite somebody to the group with or without a vote. Also how we are handling error notification globally. This screen probably be redesigned to show how many members have accepted and have been invited but when this was created there was no join screen and no way to gather this data. Now there is. I'll put a relevant TODO